nvsvocprez/utils.py

```diff
@@ -250,4 +253,92 @@ def get_alt_profiles() -> Dict:
\ No newline at end of file
+        return {}   # Return blank list to avoid internal server error.
+    
+    
+    
+def get_alt_profile_objects(collection: Dict, alt_profiles: Dict) -> Dict:
+    """Generate Profile objects for all alt profiles.
+    
+    Args:
+        collection (Dict): Dict representing collection data.
+        alt_profiles(Dict): Dict of alt profiles { uri : {profile_data}, ...}.
+        
+    Returns:
+        Dict: Dict of Profile objects representing each alternate profile.
+            { profile_name: ProfileObject, ... }
+    """
+    profiles = {}
+    for url, alt in alt_profiles.items():
+        if (
+            url in collection["conforms_to"]["value"]
+        ):
+            p = Profile(
+                uri=url,
+                id=alt['token'],
+                label=alt['name'],
+                comment=alt['vocprezdesc'],
+                mediatypes=RDF_MEDIATYPES,
+                default_mediatype="text/turtle",
+                languages=["en"],
+                default_language="en",
+            )
+        profiles[alt['token']] = p
+    return profiles
+
+
+def get_collection_query(profile: str, instance_uri: str, exclude_profiles: list):
+    """Method to generate a query for the collections page excluding certain profiles.
+    
+    Args:
+        profile (str): Profile name (e.g. 'nvs', 'puv', 'iadopt').
+        insance_uri (str): Instance URI.
+        exclude_profiles: List of profile URI's to be excluded from query.
+    Returns:
+        str: The construncted sparql query.
+    """
+    filter_text = ""
+    if profile != "nvs":
+        filter_text += """
+            FILTER ( ?p2 != skos:broader )
+            FILTER ( ?p2 != skos:narrower )
+            FILTER ( ?p2 != skos:related )
+            FILTER ( ?p2 != owl:sameAs )
+        """
+    
+    for profile in exclude_profiles:
+        # Build filter text.
+        filter_text += f'FILTER (!STRSTARTS(STR(?p2), "{profile}"))\n'
+    
+    query = f"""
+        PREFIX dc: <http://purl.org/dc/terms/>
+        PREFIX dce: <http://purl.org/dc/elements/1.1/>
+        PREFIX grg: <http://www.isotc211.org/schemas/grg/>
+        PREFIX owl: <http://www.w3.org/2002/07/owl#>
+        PREFIX pav: <http://purl.org/pav/>
+        PREFIX puv: <https://w3id.org/env/puv#>
+        PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+        PREFIX void: <http://rdfs.org/ns/void#>
+        CONSTRUCT {{
+            <xxx> ?p ?o .
+            <xxx> skos:member ?m .
+            ?m ?p2 ?o2 .
+        }}
+        WHERE {{
+            {{
+            <xxx> ?p ?o .
+            MINUS {{ <xxx> skos:member ?o . }}
+            }}
+            {{
+            <xxx> skos:member ?m .
+            ?m a skos:Concept .
+            ?m ?p2 ?o2 .
+            FILTER ( ?p2 != skos:broaderTransitive )
+            FILTER ( ?p2 != skos:narrowerTransitive )
+            {filter_text}   
+            }}
+        }}
+    """.replace(
```

Comment:
would it be better to change the xxxs within the string with {instance_uri} instead of using the replace method? Don't know if there would be any functional difference, just looks odd to me having the mix